rq/cli/helpers.py

```diff
@@ -198,6 +202,82 @@ def setup_loghandlers_from_args(verbose, quiet, date_format, log_format):
     setup_loghandlers(level, date_format=date_format, log_format=log_format)
 
 
+def job_func(func, args, kwargs):
+    try:
+        importlib.import_module(func)  # execute file
+    except ImportError:
+        return import_attribute(func)(*args, **kwargs)
+
+
+def parse_function_arg(argument):
+    mode = 0  # 0: text, 1: json, 2: literal_eval (python)
```

Comment:
We should just remove modes 1 (plain text) and 2 (json). We shouldn't need to give other options, since we're executing Python code, using `literal_eval()` makes the most sense. This should simplify both the implementation and documentation.

We just need to explain that all args/kwargs will be interpreted as Python.

Reply:
Yes we are executing python, but this cli may be called by other programms (not just shells), which have native json support. Also using strings if you are calling from shell is not very convenient. For example the literal-eval of `'abc'` is `abc` but if your just executing `echo 'test'` in your bash or zsh, you see that your shell parses this to just `test` (not `'test'`) which will throw an error.

Reply:
> Yes we are executing python, but this cli may be called by other programms (not just shells), which have native json support.

Yes but the enqueue process will be done from within Python anyway, right? So I don't think supporting JSON makes sense here as they need to be loaded from JSON string to Python objects anyway. So I'm firm on removing JSON support here as it doesn't make sense to have this layer of indirection.

> Also using strings if you are calling from shell is not very convenient. For example the literal-eval of 'abc' is abc but if your just executing echo 'test' in your bash or zsh, you see that your shell parses this to just test (not 'test') which will throw an error.

This is a good point, since this is a CLI, we want this to behave well in the CLI world. Then let's just allow two flavors, text and Python which is evaluated using `literal_eval()`.

Thanks for your input and hard work, sorry this is taking a while.

Reply:
> Yes but the enqueue process will be done from within Python anyway, right?

I don't think so. From python you wouldn't call the cli, because you have the `queue.enqueue` method.

But if you want to enqueue a function for example with node.js (or most other languages), you could make a json from your object and pass it to the cli.

Reply:
Alright then, I agree about this in principle, let me review this again to see where we can streamline things.